tests/e2e/specs/miscellaneous/CheckUserEnvsTest.spec.ts

```diff
@@ -0,0 +1,97 @@
+/** *******************************************************************
+ * copyright (c) 2020-2024 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+import { e2eContainer } from '../../configs/inversify.config';
+import { CLASSES, TYPES } from '../../configs/inversify.types';
+import { WorkspaceHandlingTests } from '../../tests-library/WorkspaceHandlingTests';
+import { ProjectAndFileTests } from '../../tests-library/ProjectAndFileTests';
+import { LoginTests } from '../../tests-library/LoginTests';
+import { registerRunningWorkspace } from '../MochaHooks';
+import { KubernetesCommandLineToolsExecutor } from '../../utils/KubernetesCommandLineToolsExecutor';
+import { ShellExecutor } from '../../utils/ShellExecutor';
+import { ITestWorkspaceUtil } from '../../utils/workspace/ITestWorkspaceUtil';
+import { ShellString } from 'shelljs';
+import { expect } from 'chai';
+import { Dashboard } from '../../pageobjects/dashboard/Dashboard';
+import { BrowserTabsUtil } from '../../utils/BrowserTabsUtil';
+import { OAUTH_CONSTANTS } from '../../constants/OAUTH_CONSTANTS';
+
+suite('"User secrets in workspace" test', function (): void {
+	const workspaceHandlingTests: WorkspaceHandlingTests = e2eContainer.get(CLASSES.WorkspaceHandlingTests);
+	const projectAndFileTests: ProjectAndFileTests = e2eContainer.get(CLASSES.ProjectAndFileTests);
+	const loginTests: LoginTests = e2eContainer.get(CLASSES.LoginTests);
+	const dashboard: Dashboard = e2eContainer.get(CLASSES.Dashboard);
+	const browserTabsUtil: BrowserTabsUtil = e2eContainer.get(CLASSES.BrowserTabsUtil);
+
+	let kubernetesCommandLineToolsExecutor: KubernetesCommandLineToolsExecutor = e2eContainer.get(
+		CLASSES.KubernetesCommandLineToolsExecutor
+	);
+	const shellExecutor: ShellExecutor = e2eContainer.get(CLASSES.ShellExecutor);
+	const testWorkspaceUtil: ITestWorkspaceUtil = e2eContainer.get(TYPES.WorkspaceUtil);
+	const userProject: string = OAUTH_CONSTANTS.TS_SELENIUM_OCP_USERNAME + `-devspaces`;
```

Comment:
The test will be supporting both Eclipse Che and Dev Spaces if use `TS_API_TEST_NAMESPACE` constant instead of `TS_SELENIUM_OCP_USERNAME`:
```
const userProject: string = API_TEST_CONSTANTS.TS_API_TEST_NAMESPACE || "admin-devspaces"
```